### DISCUSSION TITLE
Please use GitHub's "linked issue" feature

### DISCUSSION THREAD

**Discussion by FloEdelmann** [streetcomplete/StreetComplete#2917] [Repository Maintenance] [answered]
This is mainly targeted towards @westnordost and @matkoniecz, but also everyone else opening PRs:

### TL;DR

Please put `fixes #2913` in the PR description, not the PR title.

### Explanation:

You often include the issue reference in the PR title, e.g. #2914: `"does not exist" enabled for more quests, fixes #2913`.
This makes it clear for humans which issue this PR addresses.

GitHub however does only recognize this in the PR description: https://docs.github.com/en/issues/tracking-your-work-with-issues/creating-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword

In comparison: e.g. #2872 has `Fixes #1285` in the PR description, so the linked issue is recognized:
![grafik](https://user-images.githubusercontent.com/202916/119828405-6a552b00-befa-11eb-87f9-cc74a77d7f50.png)

This enables some cool features:

* When the PR is merged, the issue is closed automatically.
  Right now, it's only auto-closed if the merge commit message also includes the `fixes #2913` keyword.
* In the PR list, it shows that an issue is linked:
  ![grafik](https://user-images.githubusercontent.com/202916/119828732-c28c2d00-befa-11eb-821d-d374535d54e4.png)
* In the issue list, it shows that a PR is linked:
  ![grafik](https://user-images.githubusercontent.com/202916/119828838-dd5ea180-befa-11eb-9993-5a691ac6ef20.png)
* It allows searching for issues with / without a linked PR and vice-versa: https://docs.github.com/en/github/searching-for-information-on-github/searching-on-github/searching-issues-and-pull-requests#search-for-linked-issues-and-pull-requests
* In an issue, linked PRs are highlighted right in the header:
  <img src="https://user-images.githubusercontent.com/202916/119829972-16e3dc80-befc-11eb-81aa-dbe2614406b6.png" width="500">

**Comment by westnordost** [comment, marked as answer]
🆗

**Reply by matkoniecz** [reply]
OK

I keep forgetting about this GitHub bug (and it seems that there is no way to report bugs to GitHub at all, in past using "contact GitHub" resulted in at least placebo response)

**Reply by Lee-Carre** [reply]
@matkoniecz

> there is no way to report bugs to GitHub

There is: [github/feedback](/github/feedback).

**Reply by matkoniecz** [reply]
It seems that this bug was fixed in meantime! I was unable to reproduce it.

**Comment by FloEdelmann** [comment]
Would be okay to ask you (either the PR authors or the repo maintainers) to edit the descriptions of the following currently open PRs to include the link to the respective issue?

* ~PR #2894 by @sumanabhi: fixes #2890~
* ~PR #2856 by @Helium314: fixes #87~
* ~PR #2852 by @matkoniecz: fixes #2774~
* ~PR #2591 by @arrival-spring: fixes #2542~

**EDIT:** Everything is linked now :)

**Reply by matkoniecz** [reply]
> PR #2852 by @matkoniecz: fixes #2774

done though it is the smallest problem of that PR :)

**Reply by westnordost** [reply]
Oh, I thought you had access too. I think it would make sense. Especially if you find small things that can be enhanced etc., then you could just commit them if you don't need a review.

**Comment by rugk** [comment]
BTW you can nowadays also link issues manually _afterwards_ i.e. after the PR is already created without editing the PR body or so. That's somewhat convenient at least:
![image](https://user-images.githubusercontent.com/11966684/153933083-4748ade3-526c-4725-acbd-8c4a2cefe2f6.png)

See https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#manually-linking-a-pull-request-to-an-issue

**Reply by FloEdelmann** [reply]
I tried this now at two different PRs, but the search feature there is so bad that it doesn't let me find the relevant issues. So I resort back to editing the PR description.

**Reply by riQQ** [reply]
You can just put in the issue number (with or without a preceding #).

**Reply by FloEdelmann** [reply]
I tried that, too, but the issue I searched didn't show up in the search results.

**Reply by rugk** [reply]
Oh, this always worked for me. Of course you can only link issues of your own repo - okay, maybe that is different, but apart from that, it always worked. ANyway the syntax is there to stay, so this will always also work.

**Reply by riQQ** [reply]
I could reproduce FloEdelmann's issue but couldn't find a reason which issues are not linkable and why.
